### PR TITLE
Add OpenHarmony support to mach and CI

### PR DESCRIPTION
**Pull request by jschwe**
This allows building `servoshell` for OpenHarmony and tests this in CI.
Servoshell itself doesn't contain any OpenHarmony specific code yet, but 
still CI will help ensure that `libservo` and all dependencies compile for 
OpenHarmony. This is especially useful for gfx / fonts, since that area is
currently receiving some refactoring and CI for OpenHarmony would ensure
that the refactoring doesn't accidentally break the OpenHarmony build.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] There are tests for these changes: A CI job was added

### CODE REVIEW

File: .github/workflows/quick-check.yml
Comment (jschwe) [inline .github/workflows/quick-check.yml:1]: This change is intended to be dropped after reviewal and before merging, and is just to demonstrate that the CI workflow works. The `try` run probably won't work until the change is merged to main.

**Comment by jschwe** [comment]
Link to the ohos CI action run in my fork: https://github.com/jschwe/servo/actions/runs/9527245492

File: .github/workflows/dispatch-workflow.yml [hunk 1]
```diff
@@ -57,3 +57,12 @@ jobs:
     secrets: inherit
     with:
       profile: ${{ inputs.profile }}
+
+  OpenHarmony:
```
Comment (sagudev) [inline .github/workflows/dispatch-workflow.yml:61]:
```suggestion
  ohos:
```
to be more consistent with the rest

**Comment by sagudev** [comment]
There also need to be build-ohos in https://github.com/servo/servo/blob/main/.github/workflows/main.yml#L41 to actually run on mq.

File: .github/workflows/quick-check.yml
Comment (sagudev) [reply .github/workflows/quick-check.yml:1]: You can just drop this and provide link to workflow run that was created via `mach try ohos`

Comment (jschwe) [reply .github/workflows/quick-check.yml:1]: Thanks! Dropped the commit and update the link.

**Review by sagudev — APPROVED** [review]
Changes to workflows and and try_parser looks good to me.

@mrobinson can you review the rest?

File: .github/workflows/dispatch-workflow.yml [hunk 2]
```diff
@@ -57,3 +57,12 @@ jobs:
     secrets: inherit
     with:
       profile: ${{ inputs.profile }}
+
+  ohos:
+    if: ${{ inputs.workflow == 'ohos' }}
+    name: OpenHarmony
+    uses: ./.github/workflows/ohos.yml
+    # I believe we don't need any secrets in the OpenHarmony CI workflow.
```
Comment (mukilan) [inline .github/workflows/dispatch-workflow.yml:65]: Perhaps we can remove the commented lines.

**Review by mukilan — APPROVED** [review]

File: .github/workflows/dispatch-workflow.yml (hunk 2 above)
Comment (jschwe) [reply .github/workflows/dispatch-workflow.yml]: done

**Pull request merged**